docs/topics/data-analysis/data-analysis-work-with-api.md

```diff
@@ -1,18 +1,201 @@
+   >
+   {type="tip"}
+
+3. Create a new Kotlin Notebook by selecting **File** | **New** | **Kotlin Notebook**.
+4. In the Kotlin Notebook, import the Kotlin DataFrame library by running the following command:
+
+```kotlin
+%use dataframe
+```
+
+## Fetch data from an API
+
+Fetching data from APIs using the Kotlin Notebook with the DataFrame library is achieved through the [`.read()`](https://kotlin.github.io/dataframe/read.html) 
+function, which is similar to [retrieving data from files](data-analysis-work-with-data-sources.md#retrieve-data-from-a-file), such as CSV or JSON.
+However, when working with web-based sources, you might require additional formatting to transform the raw API data into 
+a structured format.
+
+Let's look at an example of fetching data from the [YouTube Data API](https://console.cloud.google.com/apis/library/youtube.googleapis.com):
+
+1. **Import libraries:** Import the DataFrame library, which is essential for data manipulation tasks.
+This is done by running the following command in a code cell:
+
+```kotlin
+%use dataframe
+```
+
+2. **Set up API key:** Securely add your API key in a new code cell, which is necessary for authenticating requests to the YouTube Data API. 
+You can obtain your API key from the [credentials tab](https://console.cloud.google.com/apis/credentials). :
+
+```kotlin
+val apiKey = "YOUR-API_KEY"
+```
+
+3. **Create a function to fetch and load data:** Create a load function that takes a path as a string and uses the DataFrame's `.read()` method to fetch data from the YouTube Data API:
+
+```kotlin
+fun load(path: String): AnyRow = DataRow.read("https://www.googleapis.com/youtube/v3/$path&key=$apiKey")
+```
+
+4. **Organize data into rows:** Organize the fetched data into rows and handle the YouTube API's pagination through the `nextPageToken`. 
+This ensures you gather data across multiple pages:
+
+```kotlin
+fun load(path: String, maxPages: Int): AnyFrame {
+    val rows = mutableListOf<AnyRow>()
+    var pagePath = path
+    do {
+        val row = load(pagePath)
+        rows.add(row)
+        val next = row.getValueOrNull<String>("nextPageToken")
+        pagePath = path + "&pageToken=" + next
+    } while (next != null && rows.size < maxPages)
+    return rows.concat()
+}
+```
+
+5. **Create the DataFrame:** Use the previously defined `load()` function to fetch data and create a DataFrame in a new code cell. 
+This example fetches data, or in this case, videos related to Kotlin, with a maximum of 50 results per page, up to a maximum of 5 pages. 
+The result is stored in the `df` variable:
+
+```kotlin
+val df = load("search?q=cute%20cats&maxResults=50&part=snippet", 5)
+df
+```
+
+6. **Concatenate items and display the resulting DataFrame:** Finally, extract and concatenate items from the DataFrame:
+
+```kotlin
+val items = df.items.concat()
+items
+```
+
+## Clean and refine data
```

Comment:
It would be very helpful if you add images of the structure of the data frame before and after the cleaning and refinement.

Reply:
great idea! - added images :)